homeassistant/components/zwave_js/cover.py

```diff
@@ -66,161 +66,242 @@ def async_add_cover(info: ZwaveDiscoveryInfo) -> None:
 
-
-def percent_to_zwave_tilt(value: int) -> int:
-    """Convert position in 0-100 scale to 0-99 scale.
-
-    `value` -- (int) Position byte value from 0-100.
-    """
-    if value > 0:
-        return round((value / 100) * 99)
-    return 0
-
-
-def zwave_tilt_to_percent(value: int) -> int:
-    """Convert 0-99 scale to position in 0-100 scale.
-
-    `value` -- (int) Position byte value from 0-99.
-    """
-    if value > 0:
-        return round((value / 99) * 100)
-    return 0
-
-
-class ZWaveCover(ZWaveBaseEntity, CoverEntity):
-    """Representation of a Z-Wave Cover device."""
-
-    _attr_supported_features = (
-        CoverEntityFeature.OPEN
-        | CoverEntityFeature.CLOSE
-        | CoverEntityFeature.SET_POSITION
-    )
-
-    def __init__(
+    def _set_position_values(
         self,
-        config_entry: ConfigEntry,
-        driver: Driver,
-        info: ZwaveDiscoveryInfo,
+        current_value: ZwaveValue,
+        target_value: ZwaveValue | None = None,
+        stop_value: ZwaveValue | None = None,
     ) -> None:
-        """Initialize a ZWaveCover entity."""
-        super().__init__(config_entry, driver, info)
-
-        self._stop_cover_value = (
-            self.get_zwave_value(COVER_OPEN_PROPERTY)
-            or self.get_zwave_value(COVER_UP_PROPERTY)
-            or self.get_zwave_value(COVER_ON_PROPERTY)
+        """Set values for position."""
+        self._attr_supported_features = (
+            (self._attr_supported_features or 0)
+            | CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.SET_POSITION
+        )
+        self._current_position_value = current_value
+        self._target_position_value = target_value or self.get_zwave_value(
+            TARGET_VALUE_PROPERTY, value_property_key=current_value.property_key
         )
 
-        if self._stop_cover_value:
+        if stop_value:
+            self._stop_position_value = stop_value
             self._attr_supported_features |= CoverEntityFeature.STOP
 
-        # Entity class attributes
-        self._attr_device_class = CoverDeviceClass.WINDOW
-        if self.info.platform_hint and self.info.platform_hint.startswith("shutter"):
-            self._attr_device_class = CoverDeviceClass.SHUTTER
-        if self.info.platform_hint and self.info.platform_hint.startswith("blind"):
-            self._attr_device_class = CoverDeviceClass.BLIND
+    def percent_to_zwave_position(self, value: int) -> int:
+        """Convert position in 0-100 scale to closed_value-open_value scale."""
+        return (
+            round(max(min(1, (value / 100)), 0) * self._position_range)
+            + self._fully_closed_position
+        )
+
+    def zwave_to_percent_position(self, value: int) -> int:
+        """Convert closed_value-open_value scale to position in 0-100 scale."""
+        return round(
+            ((value - self._fully_closed_position) / self._position_range) * 100
+        )
```

Comment:
May be you could user `homeassistant.util.ranged_value_to_percentage` here?

Reply:
this won't work for our use case. Using this logic, 0 in the 0-99 range converts to 1% when it should be 0%